parser/object_parse/die_parse/mod.rs

```diff
@@ -0,0 +1,88 @@
+use gimli;
+use ir;
+use traits;
+
+use super::Parse;
+
+mod item_kind;
+mod item_name;
+mod location_attrs;
+
+use self::item_kind::item_kind;
+use self::item_name::item_name;
+use self::location_attrs::DieLocationAttributes;
+
+/// This type alias is used to represent an option return value for
+/// a procedure that could return an Error.
+type FallilbleOption<T> = Result<Option<T>, traits::Error>;
+
+/// This struct represents the extra items required by the Parse trait's
+/// `parse_items` method. This is constructed by the compilation unit's
+/// own implementation of `parse_items`.
+pub struct DIEItemsExtra<'unit, R>
```

Comment:
Nitpick: Rust style is to only capitalize the first letter of an acronym inside an identifier: `DieItemsExtra`